### ISSUE TITLE
ComfyUI Integration with Custom Node Creation Bounty [$3000]

### ISSUE THREAD

**Issue by JJassonn69** [livepeer/bounties#53]
To ease the process of onboarding many more creators and enthusiasts, we aim to construct a ComfyUI [custom node](https://docs.comfy.org/essentials/custom_node_overview/) that is tightly integrated with Livepeer network for processing AI workflow commands from [ComfyUI](https://docs.comfy.org/get_started/introduction/) dashboard. Since, many creators 🧑‍🎨 rely on ComfyUI for incorporating open-source models into their workflows, this enhancement would significantly improve user experience and drive adoption. 🚀 

We are calling on the community to implement this important tool to [AI-network](https://docs.livepeer.org/ai/pipelines/overview#models-on-the-ai-subnet/) so that we can empower the creators with the open source, decentralised and cost effective access to the models on the Livepeer Network. Successfully integrating with ComfyUI by creating this custom node will provide opportunity for further utilisation of our [pipelines](https://docs.livepeer.org/ai/pipelines/overview#models-on-the-ai-subnet/) and bolster Livepeer as top network for AI infrastructure. 🔥

---

## Required Skillset

- Strong [Python](https://www.python.org/) and [Javascript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/) programming skills.
- Familiar with [ComfyUI](https://docs.comfy.org/get_started/introduction/) and its workflows.
- Basic background with developing SaaS products.

---

## Bounty Requirements

1. **Implementation:** Develop a working custom node to `ComfyUI` workflows that uses Livepeer AI pipelines for task completion. One should be able to host `ComfyUI` interface and use the custom build node to leverage Livepeer network. Create also a sample workflow that demonstrates the use of the custom node with multiple  pipelines from Livepeer together (e.g. text-to-image and image-to-video) similar to this [example](https://github.com/replicate/comfyui-replicate?tab=readme-ov-file/).
2. **Functionality:** The custom node should provide functionality that pipelines in AI-network are able to handle. It should also be modular such that it also needs to be easily extendable to other pipelines and additional functionalities and features. 
3. **Documentation and Testing:** The bounty will be considered complete with clear documentation for implementation and use. The node should be thoroughly tested across a wide range of use cases to ensure reliability, performance, and real-world applicability, making it user-friendly and encouraging adoption by creators.
4. **License**: The software should be published on a public GitHub repository under an [MIT license](https://opensource.org/license/mit).

---

## Scope Exclusions

- This bounty does **NOT** cover the hosting and providing SaaS product for ComfyUI interface. But can be handled if interested with a separate bounty. 

---

## Implementation Tips

To learn how to create a custom node for `ComfyUI`, refer to their [documentation](https://docs.comfy.org/get_started/introduction/) and explore the examples in their [GitHub repository](https://github.com/comfyanonymous/ComfyUI/). Additionally, be sure to review the Livepeer AI [API Reference documentation](https://docs.livepeer.org/ai/api-reference/text-to-image) for guidance on how to call the Livepeer AI pipelines.

---

## How to Apply

1. **Express Your Interest:** Comment on this issue to indicate your interest and explain why you're the ideal candidate for the task.
2. **Wait for Review:** Our team will review expressions of interest and select the best candidate.
5. **Get Assigned:** If selected, we'll assign the GitHub issue to you.
6. **Start Working:** Dive into your task! If you need assistance or guidance, comment on the issue or join the discussions in the `#developer-lounge` channel on our [Discord server](https://discord.gg/livepeer).
7. **Submit Your Work:** Create a pull request in the relevant repository and request a review.
8. **Notify Us:** Comment on this GitHub issue when your pull request is ready for review.
9. **Receive Your Bounty:** We'll arrange the bounty payment once your pull request is approved.
10. **Gain Recognition:** Your valuable contributions will be showcased in our project's [changelog](https://livepeer-ai.productlane.com/changelog).

Thank you for your interest in contributing to our project! 💛

> [!WARNING]
> Please wait for the issue to be assigned to you before starting work. To prevent duplication of effort, submissions for unassigned issues will not be accepted.

**Comment by brymut** [comment]
Hi @rickstaa, I'm interested in taking this on. I'm a web developer with an interest in web3 tech, with experience developing in Typescript and Python. I'm also familiar with the Livepeer network and have taken some time to familiarise myself at a basic level with the AI video. As for comfyui, I've also dabbled with it in the past and got the general gist of it. I also got to see it being used in a project on the corridor crew's cool project using it to spoof about sequel movies here: https://www.youtube.com/watch?v=PrPY_USgngg. I've had a look at the spec provided above & I believe I can take this on and would like to be assigned.

**Typescript Work**
- Designed & built a Voting Delegation Dashboard for the APWine DAO using React/Typescript/Tailwind - https://vote-delegation-dashboard.vercel.app/ 
- Browser plugin related. Added support for Zora NFTs viewing & the Celo blockchain onto the Mask.io browser plugin - https://github.com/DimensionDev/Maskbook/commit/810999f38a09df8d5cefb490b4ca475b837458bc & https://github.com/DimensionDev/Maskbook/commit/c5cfdf393baf6cf7997f66cbabc5e5952dc8cb4d 
- Contributions to the Shapeshift Web App & related KeepKey hardware wallet features. - https://github.com/shapeshift/web/commits?author=brymut 
- Integrating price data for Superfluid tokens into the Gnosis Safe web app backend - https://github.com/safe-global/safe-eth-py/pull/381 - 

**Python Work**
- Pylint plugin for improving code analysis for when using Django, forked from pydev - https://github.com/brymut/pylint-django

- You can also have a look at my Dework page: https://app.dework.xyz/profile/BryanMutai

 Ready when you are :)

**Comment by brymut** [comment]
Hi @rickstaa, maybe you could check this on this?

> Hi @JJassonn69, I'm interested in taking this on. I'm a web developer with an interest in web3 tech, with experience developing in Typescript and Python. I'm also familiar with the Livepeer network and have taken some time to familiarise myself at a basic level with the AI video. As for comfyui, I've also dabbled with it in the past and got the general gist of it. I also got to see it being used in a project on the corridor crew's cool project using it to spoof about sequel movies here: https://www.youtube.com/watch?v=PrPY_USgngg. I've had a look at the spec provided above & I believe I can take this on and would like to be assigned.
> 
> **Typescript Work**
> 
> * Designed & built a Voting Delegation Dashboard for the APWine DAO using React/Typescript/Tailwind - https://vote-delegation-dashboard.vercel.app/
> * Browser plugin related. Added support for Zora NFTs viewing & the Celo blockchain onto the Mask.io browser plugin - [DimensionDev/Maskbook@810999f](https://github.com/DimensionDev/Maskbook/commit/810999f38a09df8d5cefb490b4ca475b837458bc) & [DimensionDev/Maskbook@c5cfdf3](https://github.com/DimensionDev/Maskbook/commit/c5cfdf393baf6cf7997f66cbabc5e5952dc8cb4d)
> * Contributions to the Shapeshift Web App & related KeepKey hardware wallet features. - https://github.com/shapeshift/web/commits?author=brymut
> 
> **Python Work**
> 
> * Integrating price data for Superfluid tokens into the Gnosis Safe web app backend - [Feat: Add Superfluid Oracle token adapter  safe-global/safe-eth-py#381](https://github.com/safe-global/safe-eth-py/pull/381)
> * Pylint plugin for improving code analysis for when using Django, forked from pydev - https://github.com/brymut/pylint-django
> 
> You can also have a look at my Dework page: https://app.dework.xyz/profile/BryanMutai Discord Handle: bryanmutai
> 
> Ready when you are :)

**Comment by scapula07** [comment]
@JJassonn69 @rickstaa  hi guys. I am Eneche, is this bounty still open? if so i am very interested
I am an AI and fullstack engineer,well proficient in typescript,python and other server language. I am very familar with comfyUi and running workflows on cloud environment like comfyICU. I have tinker with other devs on ComfyUI

This is a github gist from my last session on image generation https://gist.github.com/KotaiSoen/074575a381b83ee9c327ac01b368f255

To add i am very familar with the AI subnet API and have been experimenting with it for a while since it inception.

**Comment by rickstaa** [comment]
@brymut Thank you for your interest in contributing to our open-source ecosystem! 🚀 I’ve reviewed your profile and am confident that you’re well-suited for this task. I’ve gone ahead and assigned the issue to you. If you have any questions or need any assistance, feel free to reach out to me on Discord or here.

**Comment by rickstaa** [comment]
@brymut how is your bounty hunt going? Is there anything we can do to help you out 🫡.

**Comment by RUFFY-369** [comment]
Hi @brymut, It would really be great if you could update on the current state of your `ComfyUI Integration` work as the community is really interested to get this done.
As bounty hunts are taking a long time without much regular updates so it would be really helpful if you could provide an update regarding the work on your part.
Cheers! :smile:

**Comment by brymut** [comment]
Hi @RUFFY-369, @rickstaa. I apologise for the delay, unfortunately some personal conflicts came up back I'm back on it. Will update within 24 hours.

**Comment by Nightrider0098** [comment]
Hi @RUFFY-369, @rickstaa
if @brymut is not able to work on this issue can I start working, I am a web developer who has experience in JS and Python.
I worked on module federation, Server Actions, and Live streaming. I have more than three years of experience in development.

**Comment by rickstaa** [comment]
Hi @brymut,

We wanted to inform you that the core team has partially implemented this feature for real-time AI: [livepeer-ai.productlane.com/changelog](https://livepeer-ai.productlane.com/changelog).

Could you please share an overview of the work you’ve done on the batch AI API call nodes so far? This will help us assess whether the remaining scope is still distinct enough or if we can compensate you for the work completed so far and close the bounty.

Thank you!

**Comment by rickstaa** [comment]
> Hi @RUFFY-369, @rickstaa if @brymut is not able to work on this issue can I start working, I am a web developer who has experience in JS and Python. I worked on module federation, Server Actions, and Live streaming. I have more than three years of experience in development.

Hey @Nightrider0098,

Thank you for your interest in our bounty program! Unfortunately, this particular bounty has already been assigned. But don’t worry — we’ll be posting more bounties next year. Stay tuned!

**Comment by rickstaa** [comment]
> Hi [@brymut](https://github.com/brymut),
> 
> We wanted to inform you that the core team has partially implemented this feature for real-time AI: [livepeer-ai.productlane.com/changelog](https://livepeer-ai.productlane.com/changelog).
> 
> Could you please share an overview of the work you’ve done on the batch AI API call nodes so far? This will help us assess whether the remaining scope is still distinct enough or if we can compensate you for the work completed so far and close the bounty.
> 
> Thank you!

Hi @brymut,

We’d greatly appreciate it if you could share an overview of the work you’ve completed so far, so we can move forward with compensating you for your time and effort 🙏🏻. If we don’t hear back from you within 15 days, we’ll assume the bounty was abandoned and proceed with closing it.

Thank you for your understanding!

**Comment by nishantkluhera** [comment]
Hello! I would like to work on this if it's not been done yet.

**Comment by rickstaa** [comment]
> Hello! I would like to work on this if it's not been done yet.

Hey @nishantkluhera thanks for your interest in this bounty. As explained [above](https://github.com/livepeer/bounties/issues/53#issuecomment-2606951056) this bounty was stale and currently is no longer needed. Since we didn't hear back from @brymut anymore we are closing this bounty due to inactivity. We will create more bounties related to ComfyUI soon 🔥.
